### PR TITLE
Add spark helpers

### PR DESCRIPTION
**Pull request by fhoering**

### CODE REVIEW

File: cluster_pack/spark/spark_config_builder.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def _add_file(ssb: SparkSession.Builder, path):
+    _add_or_merge(ssb, "spark.yarn.dist.archives", path)
+
+
+def _add_or_merge(ssb: SparkSession.Builder, key: str, value: str):
+    if key in ssb._options:
+        old_value = ssb._options[key]
+        ssb.config(key, ",".join(old_value.split(",") + [value]))
```
Comment (t-henri) [inline cluster_pack/spark/spark_config_builder.py]: Why not ",".join(old_value, value)

File: cluster_pack/spark/spark_config_builder.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def add_script(ssb: SparkSession.Builder, archive: str):
+    if archive.endswith('pex'):
+        _add_or_merge(ssb, "spark.yarn.dist.files", f"{archive}")
+        ssb.config("spark.executorEnv.PEX_ROOT", "./.pex")
+        os.environ['PYSPARK_PYTHON'] = './' + archive.split('/')[-1]
+        os.environ['PYSPARK_DRIVER_PYTHON'] = archive.split('/')[-1]
+    else:
+        _add_or_merge(ssb, "spark.yarn.dist.archives", f"{archive}#condaenv")
```
Comment (t-henri) [inline cluster_pack/spark/spark_config_builder.py]: You could use _add_file here

File: cluster_pack/spark/spark_config_builder.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        os.environ['PYSPARK_PYTHON'] = f"./condaenv/bin/python"
+        os.environ['PYSPARK_DRIVER_PYTHON'] = 'python'
+
+
+def add_files(ssb: SparkSession.Builder):
```
Comment (t-henri) [inline cluster_pack/spark/spark_config_builder.py]: The name is not very intuitive, could we use something like 'add_editable_requirements' ?

File: cluster_pack/spark/spark_config_builder.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import pyspark
+from pyspark.sql import SparkSession
+
+from cluster_pack import packaging, uploader
+
+from typing import Optional
+
+
+def add_script(ssb: SparkSession.Builder, archive: str):
```
Comment (t-henri) [inline cluster_pack/spark/spark_config_builder.py]: add_script doesn't really reflect what the function does, could we use something like 'add_packaged_environment' instead ?

File: cluster_pack/spark/spark_config_builder.py (hunk 1 above)
Comment (fhoering) [reply cluster_pack/spark/spark_config_builder.py]: The old value might already be a list:

File: cluster_pack/spark/spark_config_builder.py (hunk 2 above)
Comment (fhoering) [reply cluster_pack/spark/spark_config_builder.py]: Done

File: cluster_pack/spark/spark_config_builder.py (hunk 3 above)
Comment (fhoering) [reply cluster_pack/spark/spark_config_builder.py]: Done

File: cluster_pack/spark/spark_config_builder.py (hunk 4 above)
Comment (fhoering) [reply cluster_pack/spark/spark_config_builder.py]: Done

**Pull request merged**